### PR TITLE
Display progress for individual mission items

### PR DESCRIPTION
**Pull request by nexton-winjeel**
A proposed change in MAVLink adds mission item progress to the `MISSION_CURRENT` message: https://github.com/mavlink/mavlink/pull/2029

This PR adds support for displaying this information on the FlyView map:
qgc-item-progress-distance
qgc-item-progress-count
qgc-item-progress-time

Associated ArduPilot PR: https://github.com/ArduPilot/ardupilot/pull/25249

**NOTE:** This PR won't build as-is, because it doesn't include the changes to the `MISSION_CURRENT` message definition.

### CODE REVIEW

**Comment by Davidsastresas** [comment]
This is really cool, thanks.

I think we should wait until the correspondent commits in mavlink and Ardupilot are approved, so we can update mavlink right away and make it work.

I haven't taken a deep look, but I noticed a typo in src/MissionManager/SimpleMissionItem.cc, "altitide" instead of "altitude".

Would you mind pinging me again when the PRs get merged in mavlink and autopilot to look into this again? Thanks!

File: src/MissionManager/SimpleMissionItem.cc [hunk 1]
```diff
@@ -418,10 +418,26 @@ QString SimpleMissionItem::abbreviation() const
...
         return tr("ROI");
     case MAV_CMD_NAV_LOITER_TIME:
+        return tr("Loiter (time)");
     case MAV_CMD_NAV_LOITER_TURNS:
+        return tr("Loiter (turns)");
     case MAV_CMD_NAV_LOITER_UNLIM:
-    case MAV_CMD_NAV_LOITER_TO_ALT:
         return tr("Loiter");
+    case MAV_CMD_NAV_LOITER_TO_ALT:
+        return tr("Loiter (altitide)");
```
Comment (nexton-winjeel) [inline src/MissionManager/SimpleMissionItem.cc:427]:
```suggestion
        return tr("Loiter (altitude)");
```

**Comment by nexton-winjeel** [comment]
> This is really cool, thanks.
> 
> I think we should wait until the correspondent commits in mavlink and Ardupilot are approved, so we can update mavlink right away and make it work.

Agreed. The consensus on the ArduPilot side is that this capability applies to more than just mission items, so will require a new message rather than an extension to `MISSION_CURRENT`.

> I haven't taken a deep look, but I noticed a typo in src/MissionManager/SimpleMissionItem.cc, "altitide" instead of "altitude".

Yep, thanks for that. I've fixed it, I just haven't pushed it because I know the builds will fail.

> Would you mind pinging me again when the PRs get merged in mavlink and autopilot to look into this again? Thanks!

Will do. Thanks!

**Comment by ryanjAA** [comment]
Thanks! This is a great addition.

**Comment by DonLakeFlyer** [comment]
You can reopen this once it's dependent parts (mavlink, ...) get completed.

**Pull request closed without merging**